resources/services/ec2/vpc_endpoint_services.go

```diff
@@ -0,0 +1,167 @@
+				Type:        schema.TypeStringArray,
+				Resolver:    resolveEc2VpcEndpointServicePrivateDnsNames,
+			},
+			{
+				Name:        "id",
+				Description: "The ID of the endpoint service.",
+				Type:        schema.TypeString,
+				Resolver:    schema.PathResolver("ServiceId"),
+			},
+			{
+				Name:        "service_name",
+				Description: "The Amazon Resource Name (ARN) of the service.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "service_type",
+				Description: "The type of service.",
+				Type:        schema.TypeStringArray,
+				Resolver:    resolveEc2VpcEndpointServiceServiceType,
+			},
+			{
+				Name:        "tags",
+				Description: "Any tags assigned to the service.",
+				Type:        schema.TypeJSON,
+				Resolver:    resolveEc2VpcEndpointServiceTags,
+			},
+			{
+				Name:        "vpc_endpoint_policy_supported",
+				Description: "Indicates whether the service supports endpoint policies.",
+				Type:        schema.TypeBool,
+			},
+		},
+	}
+}
+
+// ====================================================================================================================
+//                                               Table Resolver Functions
+// ====================================================================================================================
+func fetchEc2VpcEndpointServices(ctx context.Context, meta schema.ClientMeta, _ *schema.Resource, res chan<- interface{}) error {
+	var config ec2.DescribeVpcEndpointServicesInput
+	c := meta.(*client.Client)
+	svc := c.Services().EC2
+	for {
+		output, err := svc.DescribeVpcEndpointServices(ctx, &config, func(options *ec2.Options) {
+			options.Region = c.Region
+		})
+		if err != nil {
+			return diag.WrapError(err)
+		}
+		res <- output.ServiceDetails
+		if aws.ToString(output.NextToken) == "" {
+			break
+		}
+		config.NextToken = output.NextToken
+	}
+	return nil
+}
+func resolveEc2VpcEndpointServicePrivateDnsNames(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ServiceDetail)
+	pdn := make([]string, 0, len(r.PrivateDnsNames))
+	for _, n := range r.PrivateDnsNames {
+		pdn = append(pdn, *n.PrivateDnsName)
+	}
+	return diag.WrapError(resource.Set("private_dns_names", pdn))
+}
+func resolveEc2VpcEndpointServiceServiceType(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ServiceDetail)
+	st := make([]string, 0, len(r.ServiceType))
+	for _, std := range r.ServiceType {
+		st = append(st, string(std.ServiceType))
+	}
+	return diag.WrapError(resource.Set("service_type", st))
+}
+func resolveEc2VpcEndpointServiceTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
+	r := resource.Item.(types.ServiceDetail)
+	tags := map[string]*string{}
+	for _, t := range r.Tags {
+		tags[*t.Key] = t.Value
+	}
+	return diag.WrapError(resource.Set("tags", tags))
+}
```

Comment:
There is a client utility that can do most of this for you:
```suggestion
func resolveEc2VpcEndpointServiceTags(ctx context.Context, meta schema.ClientMeta, resource *schema.Resource, c schema.Column) error {
	r := resource.Item.(types.ServiceDetail)
	return diag.WrapError(resource.Set(c.Name, client.TagsToMap(r.Tags)))
}
```